app/src/main/java/com/f0x1d/logfox/ui/fragment/RecordingsFragment.kt

```diff
@@ -101,36 +102,54 @@ class RecordingsFragment: BaseViewModelFragment<RecordingsViewModel, FragmentRec
             adapter.submitList(it)
         }
 
-        viewModel.recordingStateData.observe(viewLifecycleOwner) {
-            when (it) {
-                RecordingState.IDLE -> {
-                    binding.recordFab.setImageResource(R.drawable.ic_recording)
-                    binding.recordFab.isEnabled = true
-
-                    binding.pauseFab.hide()
+        viewModel.recordingStateData.observe(viewLifecycleOwner) { state ->
+            binding.recordFab.apply {
+                val description: String
+                when (state) {
+                    RecordingState.IDLE,
+                    RecordingState.SAVING -> {
+                        setImageResource(R.drawable.ic_recording)
+                        description = getString(R.string.record)
+                        isEnabled = state == RecordingState.IDLE
+                    }
+
+                    RecordingState.RECORDING,
+                    RecordingState.PAUSED -> {
+                        setImageResource(R.drawable.ic_stop)
+                        description = getString(R.string.stop)
+                        isEnabled = true
+                    }
                 }
-
-                RecordingState.RECORDING -> {
-                    binding.recordFab.setImageResource(R.drawable.ic_stop)
-                    binding.recordFab.isEnabled = true
-
-                    binding.pauseFab.setImageResource(R.drawable.ic_pause)
-                    binding.pauseFab.show()
+                description.also {
```

Comment:
Let's better move this body inside `also` to an extension. Create `FABExtensions.kt` and place it there: something like `setDescription`. Then use this extension function in `when` branches. It will look better without this `description` variable